src/clients/c++/library/request_http.cc

```diff
@@ -1351,24 +1344,12 @@ InferHttpContextImpl::AsyncRun(
 
 Error
 InferHttpContextImpl::GetAsyncRunResults(
-    ResultMap* results, bool* is_ready,
-    const std::shared_ptr<Request>& async_request, bool wait)
+    const std::shared_ptr<Request>& async_request, ResultMap* results)
 {
-  Error err = IsRequestReady(async_request, is_ready, wait);
-  if (!err.IsOk() || !(*is_ready)) {
-    return err;
-  }
-
   std::shared_ptr<HttpRequestImpl> http_request =
       std::static_pointer_cast<HttpRequestImpl>(async_request);
 
-  {
-    std::lock_guard<std::mutex> lock(mutex_);
-    ongoing_async_requests_.erase(http_request->RunIndex());
-    curl_multi_remove_handle(multi_handle_, http_request->easy_handle_);
-  }
-
-  err = UpdateStat(http_request->Timer());
+  Error err = UpdateStat(http_request->Timer());
```

Comment:
This should happen in AsyncTransfer when the response is received and REQUEST_END timestamp is captured.

Reply:
Moved